packages/wasm/crate/tests/test_build.rs

```diff
@@ -1,577 +1,374 @@
-            full_viewing_key.encode_to_vec().as_slice(),
-            plan_slice,
-            &witness_data,
-            &authorization_data,
-        )
+    let tx_last_forgotten: Forgotten = mock_db
+        .get(&tables.tree_last_forgotten, "last_forgotten")
+        .await
+        .unwrap()
         .unwrap();
-        console_log!("Serial transaction is: {:?}", serial_transaction);
+    let tx_tree_commitments: StoreCommitment = mock_db
+        .get(&tables.tree_commitments, "tree_commitments")
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Reconstruct SCT struct.
+    let vec_store_commitments: Vec<StoreCommitment> = vec![tx_tree_commitments.clone()];
+    let sct = StoredTree {
+        last_position: Some(tx_last_position.clone()),
+        last_forgotten: Some(tx_last_forgotten),
+        hashes: [].to_vec(),
+        commitments: vec_store_commitments,
+    };
 
-        // Deserialize transactions and stringify actions in the transaction body into JSON
-        let serial_result = Transaction::decode(&*serial_transaction).unwrap();
-        let parallel_result = Transaction::decode(&*parallel_transaction).unwrap();
-        let serial_json = serde_json::to_string(&serial_result.transaction_body.actions).unwrap();
-        let parallel_json =
-            serde_json::to_string(&parallel_result.transaction_body.actions).unwrap();
+    // Convert SCT to `JsValue`.
+    let sct_json = serde_wasm_bindgen::to_value(&sct).unwrap();
 
-        // Perform assertion check
-        assert_eq!(serial_json, parallel_json);
+    // Generate witness data from SCT and specific transaction plan.
+    let witness_data_bytes: Vec<u8> = witness(plan_slice, sct_json).unwrap();
+    let witness_data: WitnessData = WitnessData::decode(&*witness_data_bytes).unwrap();
+
+    // 4. Build the serial and parallel transactions
+
+    let mut actions: Vec<Action> = Vec::new();
+    for i in transaction_plan.clone().actions.clone() {
+        if let ActionPlan::Spend(ref _spend_plan) = i {
+            let action = build_action_inner(
+                transaction_plan.clone(),
+                i.clone(),
+                full_viewing_key.clone(),
+                witness_data.clone(),
+            )
+            .unwrap();
+            actions.push(action);
+        }
+        if let ActionPlan::Output(ref _output_plan) = i {
+            let action = build_action_inner(
+                transaction_plan.clone(),
+                i,
+                full_viewing_key.clone(),
+                witness_data.clone(),
+            )
+            .unwrap();
+            actions.push(action);
+        }
     }
+
+    // Execute parallel spend transaction and generate proof.
+    let parallel_transaction = build_parallel_inner(
+        actions,
+        transaction_plan.clone(),
+        witness_data.clone(),
+        authorization_data.clone(),
+    )
+    .unwrap();
+    console_log!("Parallel transaction is: {:?}", parallel_transaction);
```

Comment:
nit: we probably don't want these console logs right?

Reply:
removed